### ISSUE TITLE
Successful metadata queries not logged

### ISSUE THREAD

**Issue by cberner** [prestodb/presto#7304]
`DataDefinitionExecution` queries that succeed are not logged. I bisect'ed this to c4c40c43b0647f983bf69096df7ee54f146fdad3. It can be reproduced with the following test:

```
public class TestQueryLogging
{
    @Test
    public void test()
            throws Exception
    {
        try (DistributedQueryRunner queryRunner = RaptorQueryRunner.createRaptorQueryRunner(ImmutableMap.of(), false, false)) {
            queryRunner.execute("CREATE TABLE foo (x bigint)");
            queryRunner.execute("DROP TABLE foo");

            QueryManager queryManager = queryRunner.getCoordinator().getQueryManager();
            assertEquals(((SqlQueryManager) queryManager).getStats().getCompletedQueries().getTotalCount(), 2);
        }
    }
}
```

**Comment by haozhun** [comment]
This might have to do with the issue that certain metadata query are not expired.

**Comment by highker** [comment]
diff out (https://github.com/prestodb/presto/pull/7337)

**Comment by nezihyigitbasi** [comment]
Fixed thru #7362
